manifests/input.yaml

```diff
@@ -56,6 +56,11 @@ work/atmosphere/INPUT/STASHmaster/STASHsections_A:
   hashes:
     binhash: 9401083e4199439b77254fc490b6597d
     md5: 8525fd107dbea6184db74089f1d55fd9
+work/atmosphere/INPUT/STASHmaster/manifest.yaml:
+  fullpath: /g/data/vk83/configurations/inputs/access-esm1p5/share/atmosphere/stash/2020.05.19/STASHmaster/manifest.yaml
+  hashes:
+    binhash: 0c5b7b98f44c8244eb5e3546ddd14ffc
+    md5: a0ed46f7a89e54ec91453533d4b3552e
```

Comment:
Damn. We're picking up the `manifest.yaml` files we generated to track the input files. 

It's not the end of the world, but it would be better if they weren't included, as they're not required for the model to run.

A possible solution would be enumerating all the stashfiles, though this would be quite verbose.

Reply:
I should have mentioned, these also make their way into the `manifests/restart.yaml` file:

```
...
work/atmosphere/manifest.yaml:
  fullpath: /g/data/vk83/configurations/inputs/access-esm1p5/modern/historical/restart/atmosphere/manifest.yaml
  hashes:
    binhash: 0268ba3a83533e93f365d40038cd7df1
    md5: 9a73dceb4812d34cf35fe316cb3ee382
```

Possibly a dumb/hacky idea, but would it break anything if we made all the `manifest.yaml` files hidden? e.g. `.manifest.yaml`. Payu doesn't seem to pick them up.

Otherwise would there be any harm if we were to make payu explicitly ignore files named `manifest.yaml`?

Reply:
> Possibly a dumb/hacky idea, but would it break anything if we made all the `manifest.yaml` files hidden? e.g. `.manifest.yaml`. Payu doesn't seem to pick them up.

It's not a bad idea, but it has the advantage, and disadvantage, of not being visible when users to a simple `ls`.

> Otherwise would there be any harm if we were to make payu explicitly ignore files named manifest.yaml?

I'd prefer to have user configurable options to specify files to ignore, which could default to include `manifest.yaml`, but could be overridden. 

Unfortunately this would require a `payu` code change, and a deployment, which is going to take time and older versions of `payu` will still pick them up for any user which clones the configuration.

The best current solution would be to add all the stashfiles explicitly, but that doesn't help with the restart manifest.

Bugger. 

Any thoughts @CodeGat?

Reply:
The `.manifest.yaml` solution is looking quite attractive ...

Reply:
> I think the .manifest.yaml solution is the best - is a manifest file useful to a regular user, anyway?

Yeah I think I agree but was aware this might end up being a lot of work for you.

Reply:
For the UM I think we would be able to point to the `STASHmaster_A` file directly, but I don't think we'd be able to point to the `/g/data/vk83/configurations/inputs/access-esm1p5/share/atmosphere/stash/2020.05.19/stasets` directory containing several cryptic files:

```
├── stasets
│   ├── manifest.yaml
│   ├── X01001218
│   ├── X01002207
│   ├── X01003236
│   ├── X01003237
│   ├── X01003254
│   ├── X01003255
│   ├── X01003274
│   ├── X01003275
│   ├── X01003276
│   ├── X01003277
│   ├── X01003278
│   ├── X01003279
│   ├── X01003280
│   ├── X01003281
│   ├── X01003286
│   ├── X01005207
│   ├── X01005208
│   ├── X01005222
│   ├── X01005223
│   └── X01010206
...
```
It looks like the UM wants this as a directory rather than a set of files in the input directory: https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/07c29292929b2a4c36ed7acb2936fe522dba204d/atmosphere/um_env.yaml#L37

Reply:
Don't let good solutions be held back by workloads :D at any rate, it shouldn't be a big change, I reckon!

Reply:
> The most annoying part, potentially, is switching all the manifest.yaml files to .manifest.yaml files on the remote :D but that shouldn't be too bad.

```
find /g/data/vk83/configurations/inputs -name 'manifest.yaml' -exec bash -c 'git mv {} $(dirname {})"/.manifest.yaml" ' \;
git commit -m 'Move to invisible manifests'
```

NOTE: Should test first with `echo` or similar

Reply:
Solved the issue via https://github.com/ACCESS-NRI/model-config-inputs/pull/18

Reply:
The files on `gadi` have been moved to other hidden paths @CodeGat? If so you can regenerate the manifests @blimlim.

Reply:
Awesome thank you @CodeGat, looks like it's worked!